Your fork introduces alot of mixed indentation.

```
pyHS100/__init__.py:1:1: W391 blank line at end of file
pyHS100/pyHS100.py:1:80: E501 line too long (87 > 79 characters)
pyHS100/pyHS100.py:1:88: W291 trailing whitespace
pyHS100/pyHS100.py:3:80: E501 line too long (87 > 79 characters)
pyHS100/pyHS100.py:4:80: E501 line too long (94 > 79 characters)
pyHS100/pyHS100.py:58:11: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:61:11: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:97:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:98:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:101:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:107:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:107:80: E501 line too long (81 > 79 characters)
pyHS100/pyHS100.py:109:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:112:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:115:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:121:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:121:80: E501 line too long (81 > 79 characters)
pyHS100/pyHS100.py:123:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:126:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:127:1: W293 blank line contains whitespace
pyHS100/pyHS100.py:129:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:132:80: E501 line too long (90 > 79 characters)
pyHS100/pyHS100.py:135:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:138:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:140:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:143:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:144:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:146:36: E251 unexpected spaces around keyword / parameter equals
pyHS100/pyHS100.py:146:38: E251 unexpected spaces around keyword / parameter equals
pyHS100/pyHS100.py:146:74: E251 unexpected spaces around keyword / parameter equals
pyHS100/pyHS100.py:146:76: E251 unexpected spaces around keyword / parameter equals
pyHS100/pyHS100.py:146:80: E501 line too long (107 > 79 characters)
pyHS100/pyHS100.py:147:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:148:1: W293 blank line contains whitespace
pyHS100/pyHS100.py:150:80: E501 line too long (94 > 79 characters)
pyHS100/pyHS100.py:151:80: E501 line too long (96 > 79 characters)
pyHS100/pyHS100.py:154:80: E501 line too long (90 > 79 characters)
pyHS100/pyHS100.py:155:80: E501 line too long (94 > 79 characters)
pyHS100/pyHS100.py:157:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:160:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:160:80: E501 line too long (119 > 79 characters)
pyHS100/pyHS100.py:162:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:165:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:167:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:169:11: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:171:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:173:38: E251 unexpected spaces around keyword / parameter equals
pyHS100/pyHS100.py:173:40: E251 unexpected spaces around keyword / parameter equals
pyHS100/pyHS100.py:174:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:177:80: E501 line too long (94 > 79 characters)
pyHS100/pyHS100.py:180:80: E501 line too long (90 > 79 characters)
pyHS100/pyHS100.py:181:80: E501 line too long (88 > 79 characters)
pyHS100/pyHS100.py:183:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:186:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:186:80: E501 line too long (94 > 79 characters)
pyHS100/pyHS100.py:188:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:191:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:193:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:195:11: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:197:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:198:1: W293 blank line contains whitespace
pyHS100/pyHS100.py:200:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:207:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:210:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:212:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:214:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:218:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:220:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:222:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:225:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:227:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:228:80: E501 line too long (105 > 79 characters)
pyHS100/pyHS100.py:231:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:232:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:233:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:233:23: W291 trailing whitespace
pyHS100/pyHS100.py:238:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:241:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:244:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:246:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:247:80: E501 line too long (105 > 79 characters)
pyHS100/pyHS100.py:248:80: E501 line too long (81 > 79 characters)
pyHS100/pyHS100.py:250:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:253:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:253:16: W291 trailing whitespace
pyHS100/pyHS100.py:254:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:255:7: E111 indentation is not a multiple of four
pyHS100/pyHS100.py:255:23: W291 trailing whitespace
pyHS100/pyHS100.py:257:21: W291 trailing whitespace
```

--- a/homeassistant/components/switch/tplink.py
+++ b/homeassistant/components/switch/tplink.py
@@ -12,8 +12,8 @@ from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
 from homeassistant.const import (CONF_HOST, CONF_NAME)
 import homeassistant.helpers.config_validation as cv
 
-REQUIREMENTS = ['https://github.com/kirichkov/pyHS100/archive/'
-                '6e1fedb08ee475a1b8a882a51aa09241e78ac7e8.zip#pyHS100==0.2.0']
+REQUIREMENTS = ['https://github.com/GadgetReactor/pyHS100/archive/'
+                '1f771b7d8090a91c6a58931532e42730b021cbde.zip#pyHS100==0.2.0']
 
 _LOGGER = logging.getLogger(__name__)
 

--- a/homeassistant/components/switch/tplink.py
+++ b/homeassistant/components/switch/tplink.py
@@ -47,6 +47,11 @@ class SmartPlugSwitch(SwitchDevice):
         """Initialize the switch."""
         self.smartplug = smartplug
         self._name = name
+        self._state = None
+        self._emeter_present = (smartplug.model == 110)
+        _LOGGER.debug("Setting up TP-Link Smart Plug HS%i", smartplug.model)
+        # Set up emeter cache
+        self._emeter_params = {}
 
     @property
     def name(self):
@@ -56,7 +61,7 @@ class SmartPlugSwitch(SwitchDevice):
     @property
     def is_on(self):
         """Return true if switch is on."""
-        return self.smartplug.state == 'ON'
+        return self._state == 'ON'
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
@@ -69,23 +74,20 @@ class SmartPlugSwitch(SwitchDevice):
     @property
     def device_state_attributes(self):
         """Return the state attributes of the device."""
-        _LOGGER.debug("Updating TP-Link energy meter data")
+        return self._emeter_params
 
-        emeter_readings = self.smartplug.get_emeter_realtime()
+    def update(self):
+        """Update the TP-Link switch's state"""
+        self._state = self.smartplug.state
 
-        if emeter_readings is False:
-            return {}
+        if self._emeter_present:
+            emeter_readings = self.smartplug.get_emeter_realtime()
 
-        current_consumption = "%.1f W" % emeter_readings["power"]
-        current = "%.1f A" % emeter_readings["current"]
-        voltage = "%.2f V" % emeter_readings["voltage"]
-        total_consumption = "%.2f kW" % emeter_readings["total"]
-
-        attrs = {
-            ATTR_CURRENT_CONSUMPTION: current_consumption,
-            ATTR_TOTAL_CONSUMPTION: total_consumption,
-            ATTR_VOLTAGE: voltage,
-            ATTR_CURRENT: current
-        }
-
-        return attrs
+            self._emeter_params[ATTR_CURRENT_CONSUMPTION] \
+                = "%.1f W" % emeter_readings["power"]
+            self._emeter_params[ATTR_TOTAL_CONSUMPTION] \
+                = "%.2f kW" % emeter_readings["total"]
+            self._emeter_params[ATTR_VOLTAGE] \
+                = "%.2f V" % emeter_readings["voltage"]
+            self._emeter_params[ATTR_CURRENT] \
+                = "%.1f A" % emeter_readings["current"]

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -144,6 +144,9 @@ hikvision==0.4
 # homeassistant.components.light.flux_led
 https://github.com/Danielhiversen/flux_led/archive/0.7.zip#flux_led==0.8
 
+# homeassistant.components.switch.tplink
+https://github.com/GadgetReactor/pyHS100/archive/1f771b7d8090a91c6a58931532e42730b021cbde.zip#pyHS100==0.2.0
+
 # homeassistant.components.switch.dlink
 https://github.com/LinuxChristian/pyW215/archive/v0.3.5.zip#pyW215==0.3.5
 
@@ -181,9 +184,6 @@ https://github.com/jamespcole/home-assistant-nzb-clients/archive/616cad591540925
 
 # homeassistant.components.qwikswitch
 https://github.com/kellerza/pyqwikswitch/archive/v0.4.zip#pyqwikswitch==0.4
-
-# homeassistant.components.switch.tplink
-https://github.com/kirichkov/pyHS100/archive/6e1fedb08ee475a1b8a882a51aa09241e78ac7e8.zip#pyHS100==0.2.0
 
 # homeassistant.components.media_player.russound_rnet
 https://github.com/laf/russound/archive/0.1.6.zip#russound==0.1.6

